From 5a252868d3ed0bbb969701ffdba98b60f0dd09a8 Mon Sep 17 00:00:00 2001
From: Huang Shijie <b32955@freescale.com>
Date: Tue, 8 Nov 2011 17:39:35 +0800
Subject: [PATCH 0710/1072] ENGR00161643-3 UTP : bugfix

git://git.freescale.com/imx/linux-2.6-imx.git imx_3.10.17_1.0.0_beta
commit 32d51caaa2852c4cce3061b455c6c8373b1a7c07

The origin code lossed one line.
fix it.

Signed-off-by: Huang Shijie <b32955@freescale.com>
Signed-off-by: Hongbo Zhong <hongbo.zhong@windriver.com>
---
 drivers/usb/gadget/fsl_updater.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/usb/gadget/fsl_updater.c b/drivers/usb/gadget/fsl_updater.c
index 83333d1..83dcd56 100644
--- a/drivers/usb/gadget/fsl_updater.c
+++ b/drivers/usb/gadget/fsl_updater.c
@@ -134,6 +134,7 @@ static ssize_t utp_file_write(struct file *file, const char __user *buf,
 	if (size < sizeof(uud->data))
 		return -EINVAL;
 	uud = utp_user_data_alloc(size);
+	if (uud == NULL)
 		return -ENOMEM;
 	if (copy_from_user(&uud->data, buf, size)) {
 		printk(KERN_INFO "[ %s ] copy error!\n", __func__);
-- 
1.7.5.4

